src/lib/comms/sol-lwm2m-client.c

```diff
@@ -1362,10 +1369,17 @@ handle_resource(void *data, struct sol_coap_server *server,
         goto exit;
     }
 
-    if (client->supports_access_control) {
-        r = get_server_id_by_link_addr(&client->connections, cliaddr, &server_id);
-        header_code = SOL_COAP_RESPONSE_CODE_INTERNAL_ERROR;
-        SOL_INT_CHECK_GOTO(r, < 0, exit);
+    r = get_server_id_by_link_addr(&client->connections, cliaddr, &server_id);
+    header_code = SOL_COAP_RESPONSE_CODE_INTERNAL_ERROR;
+    SOL_INT_CHECK_GOTO(r, < 0, exit);
+
+    if (path_size >= 1 &&
```

Comment:
why not: `if (!client->is_bootstrapping && obj_ctx->obj->id == SECURITY_OBJECT_ID) {

}